app/models/king.rb

```diff
@@ -1,22 +1,46 @@
 class King < Piece
+  def valid_move?(new_x, new_y)
+    # return false if is_obstructed?(new_x.to_i, new_y.to_i)
+    return true if
+      x_difference == (new_x.to_i - x_position).abs
+    y_difference = (new_y.to_i - y_position).abs
+    (x_difference <= 1) && (y_difference <= 1)
+  end
+  false
+end
 
+def obstructed_squares(new_x, new_y)
+  x_diff = (new_x - x_position)
 
-	def valid_move?(new_x, new_y)
-		#return false if is_obstructed?(new_x.to_i, new_y.to_i)
-  	return true if 
-  		x_difference = (new_x.to_i - x_position).abs
-    	y_difference = (new_y.to_i - y_position).abs
-    	(x_difference <= 1) && (y_difference <= 1)
+  case x_diff
+  when 2
+    # move is kingside castle
+    rectilinear_obstruction_array(7, new_y)
+  when -2
+    # move is queenside castle
+    rectilinear_obstruction_array?(0, new_y)
+  else
+    # otherwise king moves one space - can't be obstructed - return empty
+    return []
   end
-  	false
-end 
-  #still need to account for invalid move if King moves into check
+end
 
+# determine if king can move himself out of check
+def can_move_out_of_check?
+  start_x = x_position
+  start_y = y_position
+  success = false
+  ((x_position - 1)..(x_position + 1)).each do |x|
+    ((y_position - 1)..(y_position + 1)).each do |y|
+      update_attributes(x_position: x, y_position: y) if valid_move?(x, y)
```

Comment:
Regarding the logic of `can_move_out_of_check?`: its not only the king we need to check which can move.. some other piece could make a move and protect the king, thus saving the situation and moving the king out of check...

Reply:
pieces.valid_move? each do? or just check each coordinate on the board for both valid move and if it takes the king out of check?

Reply:
I recall Nikko visually showing this situation to me when he was teaching me chess. I'll look into how fix the wonky logic.

Reply:
I updated the king.rb but I just focused on the logic for the king in checkmate. It got a bit too complicated and out of my logic scope to deal with the other pieces. I decided to let it be the users decision regarding the other pieces. Hope that makes sense.